WordPress/Classes/ViewRelated/System/Sidebar/SidebarViewController.swift

```diff
@@ -221,21 +222,47 @@ private struct SidebarProfileContainerView: View {
     }
 }
 
-private extension BlogListViewModel {
-    /// Returns the most recent sites and mixes and mixes in the rest of the sites
-    /// until the display limimt is reached.
+extension BlogListViewModel {
+    /// Returns a list of sites to display in the sidebar, ensuring that:
+    /// 1. The current site is always included
+    /// 2. The most recent sites are included up to the display limit
+    /// 3. The sites are sorted alphabetically
     var topSites: [BlogListSiteViewModel] {
-        var topSites = recentSites.prefix(SidebarView.displayedSiteLimit)
-        var encounteredIDs = Set(topSites.map(\.id))
-        for site in allSites where !encounteredIDs.contains(site.id) {
-            if topSites.count >= SidebarView.displayedSiteLimit {
+        var displaySites = [BlogListSiteViewModel]()
+        var encounteredIDs = Set<TaggedManagedObjectID<Blog>>()
+
+        // Ensure the current site is included (if there is one)
+        if let currentSite {
+            displaySites.append(currentSite)
+            encounteredIDs.insert(currentSite.id)
+        }
+
+        // Add recent sites up to the limit
+        for site in recentSites {
```

Comment:
(nit) you should be able to merge these two loops in one:  `for site in recentSites + allSites`

Reply:
Good suggestion f802776192ae2130d511772c4414232b1cca5aea